### ISSUE TITLE
App fails to start with NPE when restore/deploy to OCP a checkpoint app image with authCache

### ISSUE THREAD

**Issue by tam512** [OpenLiberty/open-liberty#29306]
The exception in this defect is similar to https://github.com/OpenLiberty/ci.docker/issues/547 when performing checkpoint.

Deploy checkpoint app image with authCache to OCP cluster.  The app fails to start with the following errors in the log:

```
[7/29/24, 22:07:44:123 UTC] 00000037 com.ibm.ws.kernel.launch.internal.FrameworkManager           A Launching defaultServer (WebSphere Application Server 24.0.0.8/wlp-1.0.92.cl240820240725-1102) on Eclipse OpenJ9 VM, version 17.0.11+9 (en_US)
............
.....................................
[7/29/24, 22:07:44:378 UTC] 00000038 com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "java.lang.NullPointerException: Cannot invoke "java.lang.Object.getClass()" because "cachingProvider" is null io.openliberty.jcache.internal.CacheManagerServiceImpl 212" at ffdc_24.07.29_22.07.44.0.log
[7/29/24, 22:07:44:379 UTC] 00000033 .ibm.ws.webcontainer.security.WebAppSecurityCollaboratorImpl A CWWKS9112A: The web application security settings have changed. The following properties were modified: 
[7/29/24, 22:07:44:459 UTC] 00000042 com.ibm.ws.config.xml.internal.ConfigRefresher               A CWWKG0017I: The server configuration was successfully updated in 0.311 seconds.
[7/29/24, 22:07:44:381 UTC] 00000038 io.openliberty.jcache.internal.CacheServiceImpl              W CWLJC0011W: Error encountered while retrieving the LoggedOutCookieCache JCache. Will attempt to create it instead. The error was: java.lang.IllegalArgumentException: CWLJC0007E: Invalid syntax in cacheManager URI.  Cause is java.lang.NullPointerException: Cannot invoke "java.lang.Object.getClass()" because "cachingProvider" is null.
	at io.openliberty.jcache.internal.CacheManagerServiceImpl.getCacheManager(CacheManagerServiceImpl.java:216)
	at io.openliberty.jcache.internal.CacheServiceImpl.lambda$getCache$1(CacheServiceImpl.java:202)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:692)
	at io.openliberty.jcache.internal.CacheServiceImpl.getCache(CacheServiceImpl.java:179)
	at io.openliberty.jcache.internal.CacheServiceImpl$1.run(CacheServiceImpl.java:100)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:298)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:857)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Object.getClass()" because "cachingProvider" is null
	at io.openliberty.jcache.utils.CacheConfigUtil.preConfigureCacheManager(CacheConfigUtil.java:112)
	at io.openliberty.jcache.internal.CacheManagerServiceImpl.getCacheManager(CacheManagerServiceImpl.java:186)
	... 10 more
.
[7/29/24, 22:07:44:380 UTC] 0000003e io.openliberty.jcache.internal.CacheServiceImpl              W CWLJC0011W: Error encountered while retrieving the AuthCache JCache. Will attempt to create it instead. The error was: java.lang.IllegalArgumentException: CWLJC0007E: Invalid syntax in cacheManager URI.  Cause is java.lang.NullPointerException: Cannot invoke "java.lang.Object.getClass()" because "cachingProvider" is null.
	at io.openliberty.jcache.internal.CacheManagerServiceImpl.getCacheManager(CacheManagerServiceImpl.java:216)
	at io.openliberty.jcache.internal.CacheServiceImpl.lambda$getCache$1(CacheServiceImpl.java:202)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:692)
	at io.openliberty.jcache.internal.CacheServiceImpl.getCache(CacheServiceImpl.java:179)
	at io.openliberty.jcache.internal.CacheServiceImpl$1.run(CacheServiceImpl.java:100)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:298)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:857)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Object.getClass()" because "cachingProvider" is null
	at io.openliberty.jcache.utils.CacheConfigUtil.preConfigureCacheManager(CacheConfigUtil.java:112)
	at io.openliberty.jcache.internal.CacheManagerServiceImpl.getCacheManager(CacheManagerServiceImpl.java:186)
	... 10 more
.........................
.............
[7/29/24, 22:07:48:990 UTC] 00000037 com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "io.openliberty.checkpoint.internal.criu.CheckpointFailedException: CWWKC0458E: An error occurred restoring the server process with the following message: Cannot invoke "java.util.concurrent.ScheduledExecutorService.schedule(java.lang.Runnable, long, java.util.concurrent.TimeUnit)" because "this.scheduledExecutorService" is null io.openliberty.checkpoint.internal.CheckpointImpl checkpointOrExitOnFailure" at ffdc_24.07.29_22.07.48.0.log
[7/29/24, 22:07:48:992 UTC] 00000037 io.openliberty.checkpoint.internal.CheckpointImpl            E CWWKC0454E: The server failed to restore from a checkpoint with the following message: CWWKC0458E: An error occurred restoring the server process with the following message: Cannot invoke "java.util.concurrent.ScheduledExecutorService.schedule(java.lang.Runnable, long, java.util.concurrent.TimeUnit)" because "this.scheduledExecutorService" is null
[7/29/24, 22:07:48:996 UTC] 00000037 com.ibm.ws.app.manager.AppMessageHelper                      A CWWKZ0001I: Application QuickSec started in 5.214 seconds.
[7/29/24, 22:07:49:036 UTC] 00000026 com.ibm.ws.kernel.launch.internal.FrameworkManager           A CWWKE0084I: The server defaultServer is stopping because thread Restore failed, exiting... (00000062) called the method java.lang.System.exit: 
	at java.base/java.lang.System.exit(System.java:519)
	at io.openliberty.checkpoint.internal.CheckpointImpl.lambda$checkpointOrExitOnFailure$5(CheckpointImpl.java:318)
	at java.base/java.lang.Thread.run(Thread.java:857)

```

**Comment by tam512** [comment]
Collect trace _com.ibm.ws.logging.trace.specification=*=info:jcache=all:com.ibm.ws.security.*=all:com.ibm.websphere.security.*=all:io.openliberty.security.*=all:checkpoint=all_
and sent to @anjumfatima90

**Comment by tam512** [comment]
I verified that this problem is fixed on Liberty `24.0.0.9/wlp-1.0.93.cl240920240821-1903`

```
[8/22/24, 17:20:33:507 UTC] 00000037 com.ibm.ws.kernel.launch.internal.FrameworkManager           A Launching defaultServer (WebSphere Application Server 24.0.0.9/wlp-1.0.93.cl240920240821-1903) on Eclipse OpenJ9 VM, version 17.0.12+7 (en_US)
....... ............
[8/22/24, 17:20:37:880 UTC] 00000037 io.openliberty.checkpoint.internal.CheckpointImpl            A CWWKC0452I: The Liberty server process resumed operation from a checkpoint in 4.806 seconds.
[8/22/24, 17:20:37:882 UTC] 00000037 com.ibm.ws.app.manager.AppMessageHelper                      A CWWKZ0001I: Application QuickSec started in 4.809 seconds.
...................
[8/22/24, 17:20:37:942 UTC] 00000037 com.ibm.ws.kernel.feature.internal.FeatureManager            I CWWKF0008I: Feature update completed in 4.868 seconds.
[8/22/24, 17:20:37:942 UTC] 00000037 com.ibm.ws.kernel.feature.internal.FeatureManager            A CWWKF0011I: The defaultServer server is ready to run a smarter planet. The defaultServer server started in 4.868 seconds.
```
